service/core.go

```diff
@@ -65,9 +64,9 @@ type Core struct {
 	subscriptionsMu sync.RWMutex
 }
 
-func NewCore(ctx context.Context, conf Config) (*Core, error) {
-	if conf.Logger == nil {
-		conf.Logger = zap.NewNop()
+func NewCore(ctx context.Context, conf Config, logger *zap.Logger) (*Core, error) {
```

Comment:
I'd leave logger as a field in Config but don't have strong feelings about that.

Reply:
ok